internal/health/condition/check_ready_test.go

```diff
@@ -61,16 +95,112 @@ var _ = Describe("ReadyCheck", func() {
 						},
 					},
 				}
-				check := ReadyCheck(nil)
+				member1Lease := createMemberLease(member1Name, etcdNamespace, ptr.To(fmt.Sprintf("%s:%s:%s", member1ID, clusterID, druidv1alpha1.EtcdRoleLeader)))
+				member2Lease := createMemberLease(member2Name, etcdNamespace, ptr.To(fmt.Sprintf("%s:%s:%s", member2ID, clusterID, druidv1alpha1.EtcdRoleMember)))
+				member3Lease := createMemberLease(member3Name, etcdNamespace, ptr.To(fmt.Sprintf("%s:%s:%s", member3ID, clusterID, druidv1alpha1.EtcdRoleMember)))
+				existingObjects := mapToClientObjects([]*coordinationv1.Lease{member1Lease, member2Lease, member3Lease})
+				cl := testutils.CreateTestFakeClientForObjects(nil, nil, nil, nil, existingObjects)
+				check := ReadyCheck(cl)
 
 				result := check.Check(context.TODO(), etcd)
 
 				Expect(result.ConditionType()).To(Equal(druidv1alpha1.ConditionTypeReady))
 				Expect(result.Status()).To(Equal(druidv1alpha1.ConditionTrue))
 			})
 
+			It("should return that the cluster has a quorum (all members ready, with old lease format from backup-restore)", func() {
+				etcd := druidv1alpha1.Etcd{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      etcdName,
+						Namespace: etcdNamespace,
+					},
+					Spec: druidv1alpha1.EtcdSpec{Replicas: 3},
+					Status: druidv1alpha1.EtcdStatus{
+						Members: []druidv1alpha1.EtcdMemberStatus{
+							readyMember,
+							readyMember,
+							readyMember,
+						},
+					},
+				}
+				// Note that the cluster ID is not part of the HolderIdentity here, which is the old format used by backup-restore
+				member1Lease := createMemberLease(member1Name, etcdNamespace, ptr.To(fmt.Sprintf("%s:%s", member1ID, druidv1alpha1.EtcdRoleLeader)))
+				member2Lease := createMemberLease(member2Name, etcdNamespace, ptr.To(fmt.Sprintf("%s:%s", member2ID, druidv1alpha1.EtcdRoleMember)))
+				member3Lease := createMemberLease(member3Name, etcdNamespace, ptr.To(fmt.Sprintf("%s:%s", member3ID, druidv1alpha1.EtcdRoleMember)))
+				existingObjects := mapToClientObjects([]*coordinationv1.Lease{member1Lease, member2Lease, member3Lease})
+				cl := testutils.CreateTestFakeClientForObjects(nil, nil, nil, nil, existingObjects)
+				check := ReadyCheck(cl)
+
+				result := check.Check(context.TODO(), etcd)
+
+				Expect(result.ConditionType()).To(Equal(druidv1alpha1.ConditionTypeReady))
+				Expect(result.Status()).To(Equal(druidv1alpha1.ConditionTrue))
+			})
+
+			It("should return that the cluster has a split-brain (there are 2 leaders simultaneously)", func() {
```

Comment:
I think this tests need to be removed as this condition may not be the correct way to detect split-brain with surety.